paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/lookup/FileStoreLookupFunction.java

```diff
@@ -139,10 +140,17 @@ private void open() throws Exception {
 
         if (options.get(LOOKUP_CACHE_MODE) == LookupCacheMode.AUTO
                 && new HashSet<>(table.primaryKeys()).equals(new HashSet<>(joinKeys))) {
-            try {
+            if (isRemoteServiceAvailable(storeTable)) {
```

Comment:
Do you think we could add explicit lookup cache mode eg: `partial_local`, `partial_remote`. It will do good in the test. Otherwise, in IT test, we do not know whether it's  fallback to the `FULL` mode cache

Reply:
I would like to keep here simple now, until we have requirement to support specific modes.

I added a test to verify the mode choosing: [d14516f](https://github.com/apache/incubator-paimon/pull/2681/commits/d14516f8c1ddbffa03be096c1cea24381a5fcd71)